### PR TITLE
[SDK] Add a new SDK method to create account via private key literal

### PR DESCRIPTION
**Pull request by JmPotato**
## Type of Change
- [x] New feature
- [ ] Bug fix
- [ ] Breaking change
- [ ] Performance improvement
- [ ] Refactoring
- [ ] Dependency update
- [ ] Documentation update

## Which Components or Systems Does This Change Impact?
- [ ] Validator Node
- [ ] Full Node (API, Indexer, etc.)
- [ ] Move/Aptos Virtual Machine
- [ ] Aptos Framework
- [x] Aptos CLI/SDK
- [ ] Developer Infrastructure
- [ ] Other (specify)

## How Has This Been Tested?

Added a unit test named `test_create_account_from_private_key`.

## Key Areas to Review

See `LocalAccount::from_private_key` method.

## Checklist
- [x] I have read and followed the [CONTRIBUTING](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I identified and added all stakeholders and component owners affected by this change as reviewers
- [x] I tested both happy and unhappy path of the functionality
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

**Review by davidiw — APPROVED** [review]

**Review by gregnazario — APPROVED** [review]

**Comment by JmPotato** [comment]
@davidiw @gregnazario Hi, guys! It seems that I don't have permission to rerun these CI tests. Could you please help me trigger them? Thx a lot.

**Comment by gregnazario** [comment]
> @davidiw @gregnazario Hi, guys! It seems that I don't have permission to rerun these CI tests. Could you please help me trigger them? Thx a lot.

Sure, I must have missed clicking start.

**Comment by JmPotato** [comment]
> > @davidiw @gregnazario Hi, guys! It seems that I don't have permission to rerun these CI tests. Could you please help me trigger them? Thx a lot.
> 
> Sure, I must have missed clicking start.

@gregnazario I'm not quite sure how this works, but the CI still fails to run since there is no permission.

image

File: sdk/src/types.rs [hunk 1]
```diff
@@ -135,6 +135,19 @@ impl LocalAccount {
...
 
+    /// Create a new account from the given private key in hex literal.
+    pub fn from_private_key(private_key: &str, sequence_number: u64) -> Result<Self> {
+        let key = AccountKey::from_private_key(Ed25519PrivateKey::try_from(
+            hex::decode(private_key.trim_start_matches("0x"))?.as_ref(),
+        )?);
+
+        Ok(Self {
+            address: key.authentication_key().account_address(),
+            key,
```
Comment (gregnazario) [inline sdk/src/types.rs:146]: This needs to be setting `auth`, can you fix it @JmPotato?

Perhaps someone change this while it was in PR

Comment (JmPotato) [reply sdk/src/types.rs]: Fixed. PTAL.

**Review by gregnazario — APPROVED** [review]

**Pull request merged**